widget/embedded/src/hooks/useForceAutoConnect.ts

```diff
@@ -0,0 +1,37 @@
+import type { MetaState } from '../store/meta';
+import type { WalletType } from '@rango-dev/wallets-shared';
+
+import { useWallets } from '@rango-dev/wallets-react';
+import { useEffect, useRef } from 'react';
+
+type forceAutoConnectParams = {
+  loadingStatus: MetaState['loadingStatus'];
+  walletType?: WalletType;
+};
+
+export function useForceAutoConnect(params: forceAutoConnectParams): void {
+  const { loadingStatus, walletType = '' } = params;
+  const { connect, state } = useWallets();
+  const initiated = useRef(false);
+
+  const walletState = state(walletType);
+
+  useEffect(() => {
+    const shouldTryConnect =
+      loadingStatus === 'success' &&
+      walletType &&
+      walletState &&
+      walletState.installed &&
+      !walletState.connecting &&
+      !walletState.connected;
+
+    if (shouldTryConnect && !initiated.current) {
```

Comment:
I think `!initiated.current` can be buggy here. if `Wallet` has changed, it should be set to `false` again, If not, it only works for first referrer.

Reply:
fixed